agent/stats/snapshotter.go

```diff
@@ -247,22 +247,18 @@ func getStatsFromEnvoy(httpClient *retryablehttp.Client, request *retryablehttp.
 	log.Debugf("Stats request took: %vms", duration.Milliseconds())
 
 	if err != nil {
-		// Valid status code range is [100, 600)
-		if statsResponse != nil && statsResponse.StatusCode < 600 && statsResponse.StatusCode >= 100 {
```

Comment:
This isn't needed as if the error is non-nil, we can ignore the response and just return an error.